### PR TITLE
Add MndToCat and other related theorems

### PR DESCRIPTION
**Pull request by zwang123**
# Added
* `dtrucor3` as a learning process of how I understand `ax-5`.
* `idmon` and `idepi` showing identity morphisms are both monomorphisms and epimorphisms.
* Relevant theorems regarding monoids as categories
* The existence of non-thin categories where all morphisms are simultaneously monomorphisms and epimorphisms.

### CODE REVIEW

File: set.mm [hunk 1]
```diff
@@ -777824,6 +777899,211 @@ mean the category of preordered sets (classes).  However, "ProsetToCat"
...
+      GWCVRVEVSVFWCVKCVKCVKCWDWDWDSVKCNTUHUGRUIFUJURVCVIUKULUMUN $.
+
+    mndtcbas.b $e |- ( ph -> B = ( Base ` C ) ) $.
+    $( The base set of the category.  (Contributed by Zhi Wang, 22-Sep-2024.)
+       (New usage is discouraged.) $)
+    mndtcbasval $p |- ( ph -> B = { M } ) $=
+      ( cbs cfv cnx csn cop chom cotp cco cplusg ctp mndtcval cvv c1 fveq2d cdc
+      wcel wceq snex c5 catstr baseid snsstp1 strfv mp1i 3eqtr4d ) ACHIJHIDKZLZ
+      JMIDDDHINKZLZJOIDDDNDPILKZLZQZHIZBUMACUSHACDEFRUAGUMSUCUMUTUDADUEUMUSHSTT
+      UFUBLUQUMUOUGUHUNUPURUIUJUKUL $.
```
Comment (tirix) [inline set.mm:777956]: Maybe "_The base set of the category built from a monoid_" ?
This way, the comment would be self-contained.
(same remark for the other theorems)

**Review by tirix — APPROVED** [review]
Very nice!

**Comment by langgerard** [comment]
Maybe it would be necessary to add a theorem stating explicitly that given any category C, every diagonal Hom-set in C equipped with the restriction of the binary operation of composition has a structure of monoid ?

**Comment by zwang123** [comment]
> Maybe it would be necessary to add a theorem stating explicitly that given any category C, every diagonal Hom-set in C equipped with the restriction of the binary operation of composition has a structure of monoid ?



Something like this... Will do it in the next PR.
```
|- ( ph -> C e. Cat )
|- B = ( Base ` C )
|- ( ph -> X e. B )
|- H = ( Hom ` C )
|- ( X H X ) = ( Base ` M )
|- .x. = ( comp ` C )
|- ( <. X , X >. .x. X ) = ( +g ` M )

|- ( ph -> M e. Mnd )
```

**Comment by langgerard** [comment]
Yes, exactly. So that every category whose base set is a singleton will have a structure of Monoid

**Comment by langgerard** [comment]
Moreover, in the case that CAT 1  is true, so that the Hom-sets are distinct, for the structure <Hom(C), comp(C)> to be a monoid, the base B of the category C must be the singleton of a set. This is because : 
-if B is empty, then H is empty and cannot be the base set of a monoid ;
-if two distinct objects a and b are in b, then the diagonal hom-sets H(a,a) and H(b,b) are non-empty and distinct, but  if f is in H(a,a) and g is in H(b,b), with a and b distinct, there is no way to have a composite morphism for f and g in H(b,a) or g and f in H(a,b), so that comp(C) cannot be a binary internal operation on H because the composition of f and g does not exist.

File: set.mm (hunk 1 above)
Comment (zwang123) [reply set.mm]: https://github.com/metamath/set.mm/pull/4242/commits/7d71ac3e296982cf20d346f748e616c1fd762028

**Comment by benjub** [comment]
> > Maybe it would be necessary to add a theorem stating explicitly that given any category C, every diagonal Hom-set in C equipped with the restriction of the binary operation of composition has a structure of monoid ?
> 
> Something like this... Will do it in the next PR.
> 
> ```
> |- ( ph -> C e. Cat )
> |- B = ( Base ` C )
> |- ( ph -> X e. B )
> |- H = ( Hom ` C )
> |- ( X H X ) = ( Base ` M )
> |- .x. = ( comp ` C )
> |- ( <. X , X >. .x. X ) = ( +g ` M )
> 
> |- ( ph -> M e. Mnd )
> ```

This is https://us.metamath.org/mpeuni/bj-endmnd.html

**Comment by tirix** [comment]
> 
> This is https://us.metamath.org/mpeuni/bj-endmnd.html

The whole section "Monoid of endomorphisms" in your mathbox should be moved to main, IMO.

**Comment by langgerard** [comment]
I had not seen  bj-endmnd, and I am happy that we have it.
And I agree with Tirix that the whole section "Monoid of endomorphism" should be moved to main.

**Comment by zwang123** [comment]
> > > Maybe it would be necessary to add a theorem stating explicitly that given any category C, every diagonal Hom-set in C equipped with the restriction of the binary operation of composition has a structure of monoid ?
> > 
> > 
> > Something like this... Will do it in the next PR.
> > ```
> > |- ( ph -> C e. Cat )
> > |- B = ( Base ` C )
> > |- ( ph -> X e. B )
> > |- H = ( Hom ` C )
> > |- ( X H X ) = ( Base ` M )
> > |- .x. = ( comp ` C )
> > |- ( <. X , X >. .x. X ) = ( +g ` M )
> > 
> > |- ( ph -> M e. Mnd )
> > ```
> 
> This is https://us.metamath.org/mpeuni/bj-endmnd.html

Oof... I already proved the theorem. This could be a lemma for your proof to shorten it significantly. Maybe I should rename it to endmndlem.
```
    catmnd.b $e |- B = ( Base ` C ) $.
    catmnd.h $e |- H = ( Hom ` C ) $.
    catmnd.o $e |- .x. = ( comp ` C ) $.
    catmnd.c $e |- ( ph -> C e. Cat ) $.
    catmnd.x $e |- ( ph -> X e. B ) $.
    catmnd.m $e |- ( ph -> ( X H X ) = ( Base ` M ) ) $.
    catmnd.p $e |- ( ph -> ( <. X , X >. .x. X ) = ( +g ` M ) ) $.

    catmnd $p |- ( ph -> M e. Mnd ) $=
```

**Comment by benjub** [comment]
> > This is https://us.metamath.org/mpeuni/bj-endmnd.html
> 
> The whole section "Monoid of endomorphisms" in your mathbox should be moved to main, IMO.

Done in #4249.

> This [catmnd] could be a lemma for your proof to shorten it significantly. Maybe I should rename it to endmndlem.

I don't see `catmnd`, but if you can shorten a proof, then you are encouraged to do it of course !

**Review by icecream17 — APPROVED** [review]

**Pull request merged**